### ISSUE TITLE
Virtual key setup loop, cannot set up virtual key, cannot send commands

### ISSUE THREAD

**Issue by sageDieu** [itsMeDavidV/Watchla-Support#193]
I installed the latest version 3.4.9 in the app store last night, signed out, signed back in with the new method. Phone and watch can receive car data and watch as bluetooth key still works. 

However both phone and watch are unable to send commands as they now require a "virtual key" to be configured. 

Regardless of if I trigger this process from phone app or watch QR code, I am routed to what appears to be a broken URL: https://www.tesla.com/_ak/watchla.app?callback=undefined&id=undefined

This URL prompts me to download the tesla app and sign in (already done) and then tap finish setup. When I tap that button, it simply reloads the page. It seems like Watchla isn't passing through the correct parameters as when I follow the same process fresh from Tessie I'm sent into the Tesla app directly rather than my browser.

![image](https://github.com/itsMeDavidV/Watchla-Support/assets/895847/11d414a7-6dd8-45d5-8da5-80008c957761)

**Comment by itsMeDavidV** [comment]
Hi @sageDieu unfortunately the process to add a virtual key can sometimes be spotty. Tesla is working on this. 

For now, I'd recommend skipping the Finish Setup button and tapping the "Open" button on the top right of the page. For now this seems like the only solution in this scenario that works. (See attached screenshot for reference)

![image](https://github.com/itsMeDavidV/Watchla-Support/assets/12491192/bdf01eff-760b-4bfa-bd62-448414b0f030)

To answers your questions, there are no parameters as of yet for this url, the url parameters added are most likely testing by Tesla on what we hope will be a callback url feature that is soon to come. Also, apps can't directly open other apps quite like you describe. In short your iPhone decides whether to open the url or app associated with the url, and this decision is out of the control of the app that hosts the URL.

**Comment by sageDieu** [comment]
Ah nice, that seems to have worked (the open button) - I had to manually paste the link into Safari as I use a privacy-oriented browser with an extension that hides that flavor of ad/popup. Confirmed once I tapped "open" it took me to the Tesla app where I was able to approve adding watchla.app as a virtual key. Now commands from the watch work as expected. 

Thanks for the tip - I definitely wouldn't have figured that out on my own (honestly I don't think I would have pressed open even if I had seen it), so potentially could be worth adding a little tutorial for this when you go to set up a virtual key the first time (press open in the top right, then approve in the tesla app). Of course only if others are confused by the same thing :D 

Thanks for the great app and quick response here!
